spec/controllers/timetables_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe TimetablesController, type: :controller do
+
+  describe "#format_dates" do
+    it "returns an array of date objects" do
+      dates = ["2017-06-06", "2017-05-29"]
+      expect(TimetablesController.new.send(:format_dates, dates).first.class).to eql(Date)
```

Comment:
I thought you could use a `controller` variable to get at the current controller in controller specs. If that's a thing i made up I would add a this to the top and use it through out `let(:controller) { TimetablesController.new }`